rcl_action/test/rcl_action/test_graph.cpp

```diff
@@ -94,10 +94,6 @@ class CLASSNAME (TestActionGraphFixture, RMW_IMPLEMENTATION) : public ::testing:
 
     ret = rcl_shutdown(&this->context);
     EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
-    ret = rcl_context_fini(&this->context);
```

Comment:
Is there an issue with finalizing the context here? This is how it is done in the rcl graph tests as well:

https://github.com/ros2/rcl/blob/0c4c743d29d3c313a46d1079003e49081a64e292/rcl/test/rcl/test_graph.cpp#L121-L123

Reply:
You're right, there is inconsistency in calling `rcl_context_fini()` between the tests. All tests except `test_graph` seem to expect `rcl_shutdown()` to take care of freeing the `context` as well, so I went with this.

Should I rather move the call to individual tests?

Reply:
I'm pretty sure `test_graph` is handling it correctly, it is the other tests that have memory leaks (I just noticed). Thanks for finding the bug!

Reply:
To be clear, I think calls to `rcl_context_fini` should be added to the other tests and this change (and the one to `rcl_shutdown` be reverted.

Reply:
Sounds good. I'll follow up with a new commit.